app/frontend/wallet/constants.ts

```diff
@@ -96,10 +97,10 @@ export const NETWORKS = {
 
 export const DELAY_AFTER_TOO_MANY_REQUESTS = 2000
 export const ADALITE_SUPPORT_EMAIL = 'adalite@vacuumlabs.com'
-export const CRYPTO_PROVIDER_TYPES = {
-  LEDGER: 'LEDGER',
-  TREZOR: 'TREZOR',
-  WALLET_SECRET: 'WALLET_SECRET',
+export const enum CRYPTO_PROVIDER_TYPES {
```

Comment:
the `const` can give a false sense of "contantness" of the object, for enums it's generally advisable to freeze the object as well: https://medium.com/@jdgelok/javascript-enums-with-object-freeze-e3c1cca54de8

Reply:
ah sorry - that advice applies sense for JS, not for typescript :) But anyway - it appears that `const enum` may not be what we want as those are removed from the compiled JS: https://www.typescriptlang.org/docs/handbook/enums.html#const-enums - is there any strong reason to have the enums as const?

Reply:
const enum in typescript is rather a beast. I basically boils down to compile-time replacement. We generally encourage const enums unless you need to export the type ouside of the project (in which case const enum is no-go). Normal enums are rather a mess because for some inexplicable reason they figured out that enum should contain *both* forward and inverse mapping. This will shoot you in the foot if you are doing some funky stuff like `enum { a='b', b='c', c='a'}`

Reply:
I would also use CamelCaps for enum types. 
`const enum CryptoProviderType` sounds very reasonable to me in this case

Same goes for enum keys although you may want it to keep consistent with the rest of the project so maybe ALL_CAPS is ok there.

Reply:
Actually, the reason I used const enums in multiple places was that Pershing recommended it to us :D